From fdc90e21399e23d1aaeb505ab688e39432401ab9 Mon Sep 17 00:00:00 2001
From: Kai Liu <kai.liu@suse.com>
Date: Thu, 9 Jun 2022 22:38:19 +0800
Subject: [PATCH] Revert "net: hinic: add missing destroy_workqueue in
 hinic_pf_to_mgmt_init"
Patch-mainline: Never, revert stable patch that is not needed
References: bsn#83
Modified-by-SEL: No

This reverts commit 33411945c9ad8a6d58aa73806ef241ae563b0bc4.
---
 drivers/net/ethernet/huawei/hinic/hinic_hw_mgmt.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/net/ethernet/huawei/hinic/hinic_hw_mgmt.c b/drivers/net/ethernet/huawei/hinic/hinic_hw_mgmt.c
index 027dcc453506..819fa13034c0 100644
--- a/drivers/net/ethernet/huawei/hinic/hinic_hw_mgmt.c
+++ b/drivers/net/ethernet/huawei/hinic/hinic_hw_mgmt.c
@@ -647,7 +647,6 @@ int hinic_pf_to_mgmt_init(struct hinic_pf_to_mgmt *pf_to_mgmt,
 	err = alloc_msg_buf(pf_to_mgmt);
 	if (err) {
 		dev_err(&pdev->dev, "Failed to allocate msg buffers\n");
-		destroy_workqueue(pf_to_mgmt->workq);
 		hinic_health_reporters_destroy(hwdev->devlink_dev);
 		return err;
 	}
@@ -655,7 +654,6 @@ int hinic_pf_to_mgmt_init(struct hinic_pf_to_mgmt *pf_to_mgmt,
 	err = hinic_api_cmd_init(pf_to_mgmt->cmd_chain, hwif);
 	if (err) {
 		dev_err(&pdev->dev, "Failed to initialize cmd chains\n");
-		destroy_workqueue(pf_to_mgmt->workq);
 		hinic_health_reporters_destroy(hwdev->devlink_dev);
 		return err;
 	}
-- 
2.35.1

